fafbot.py

```diff
@@ -27,88 +29,47 @@
 
-        self.info = Information(TWITCH_STREAMS, GAME, STREAMER_INFO)
-        self.askForCast = 0
-        self.askForYoutube = 0
-
     def on_pubmsg(self, c, e):
         try:
-            message = e.arguments[0]
-            if message.startswith("!streams"):
-                if time.time() - self.askForCast > 60*10:
-                    self.askForCast = time.time()
-                    streams = self.info.get_game_streamer_names()
-                    try:
-                        streams_hitbox = json.loads(urllib2.urlopen(HITBOX_STREAMS).read())
-                    except:
-                        streams_hitbox = {"livestream": []}
-                    num_of_streams = len(streams["streams"]) + len(streams_hitbox["livestream"])
-                    if num_of_streams > 0:
-                        self.connection.privmsg("#aeolus", "%i Streams online :" % num_of_streams)
-                        for stream in streams["streams"]:
-                            #print stream["channel"]
-                            t = stream["channel"]["updated_at"]
-                            date = t.split("T")
-                            hour = date[1].replace("Z", "")
-
-                            self.connection.privmsg("#aeolus", "%s - %s - %s Since %s (%i viewers) " % (stream["channel"]["display_name"], stream["channel"]["status"], stream["channel"]["url"], hour, stream["viewers"]))
-                        for stream in streams_hitbox["livestream"]:
-                            self.connection.privmsg("#aeolus", "%s - %s - %s Since %s (%s viewers) " % (stream["media_display_name"], stream["media_status"], stream["channel"]["channel_link"], stream["media_live_since"], stream["media_views"]))
-                    else:
-                        self.connection.privmsg("#aeolus", "No one is streaming :'(")
-            if message.startswith("!casts"):
-                if time.time() - self.askForYoutube > 60*10:
-                    self.askForYoutube = time.time()
-                    con = urllib2.urlopen("http://gdata.youtube.com/feeds/api/videos?q=forged+alliance+-SWTOR&max-results=5&v=2&orderby=published&alt=jsonc")
-                    info = con.read()
-                    con.close()
-                    data = json.loads(info)
-                    self.connection.privmsg("#aeolus", "5 Latest youtube videos:")
-                    for item in data['data']['items']:
-                        t = item["uploaded"]
-                        date = t.split("T")[0]
-                        like = "0"
-                        if "likeCount" in item:
-                            like = item['likeCount']
-                        self.connection.privmsg("#aeolus", "%s by %s - %s - %s (%s likes) " % (item['title'], item["uploader"], item['player']['default'].replace("&feature=youtube_gdata_player", ""), date, like))
-
-
+            responses = self.message_handler(e.arguments[0])
 
+            for response in responses:
```

Comment:
Yes... how would that be better than the current approach? I don't see any real advantages or disadvantages for this code, except that using a generator might be more of a hassle for the tests (since each one will then presumably need to turn it into a list).

It is not clear to me if your comment is an FYI or something you think should be changed before merge.